File: windows/ClipSync.Core/Protocol/ProtocolEnvelope.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace ClipSync.Core.Protocol;

public sealed record ProtocolEnvelope(
    [property: JsonPropertyName("version")] int Version,
    [property: JsonPropertyName("type")] string Type,
    [property: JsonPropertyName("request_id")] Guid RequestId,
    [property: JsonPropertyName("body")] JsonElement Body);

File: android/app/src/main/java/com/clipsync/android/protocol/ProtocolEnvelope.kt
package com.clipsync.android.protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

@Serializable
data class ProtocolEnvelope(
    val version: Int,
    val type: String,
    @SerialName("request_id") val requestId: String,
    val body: JsonObject,
)
